lupa/_lupa.pyx

```diff
@@ -1207,11 +1263,17 @@ cdef int py_to_lua(LuaRuntime runtime, lua_State *L, object o, bint wrap_none=Fa
         lua.lua_pushnumber(L, <lua.lua_Number>cpython.float.PyFloat_AS_DOUBLE(o))
         pushed_values_count = 1
     elif isinstance(o, long):
-        lua.lua_pushnumber(L, <lua.lua_Number>cpython.long.PyLong_AsDouble(o))
-        pushed_values_count = 1
+        try:
+            lua.lua_pushinteger(L, <lua.lua_Integer>o)
+            pushed_values_count = 1
+        except OverflowError as e:
+            overflow_error = e
```

Comment:
I think it would be nice if we could allow users to look at the exception in `sys.exc_value`.
```suggestion
        except OverflowError:
            pushed_values_count = py_to_lua_handle_overflow(runtime, L, o)
            if pushed_values_count <= 0:
                raise
```